storage/cloudspanner/tree_storage.go

```diff
@@ -0,0 +1,523 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudspanner
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/google/trillian"
+	"github.com/google/trillian/storage"
+	"github.com/google/trillian/storage/cache"
+	"github.com/google/trillian/storage/cloudspanner/spannerpb"
+	"github.com/google/trillian/storage/storagepb"
+	"github.com/google/trillian/trees"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	// ErrNotFound is returned when a read/lookup fails because there was no such
+	// item.
+	ErrNotFound = status.Errorf(codes.NotFound, "not found")
+
+	// ErrNotImplemented is returned by any interface methods which have not been
+	// implemented yet.
+	ErrNotImplemented = errors.New("not implemented")
+
+	// ErrTransactionClosed is returned by interface methods when an operation is
+	// attempted on a transaction whose Commit or Rollback methods have
+	// previously been called.
+	ErrTransactionClosed = errors.New("transaction is closed")
+
+	// ErrWrongTXType is returned when, somehow, a write operation is attempted
+	// with a read-only transaction.  This should not even be possible.
+	ErrWrongTXType = errors.New("mutating method called on read-only transaction")
+
+	// errFinished is only used to terminate spanner reads early, once all
```

Comment:
remove spanner or rename to cloudspanner

Reply:
done